topic-controller/src/test/java/io/strimzi/test/KubeCluster.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.strimzi.test;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static java.lang.String.join;
+
+public abstract class KubeCluster {
+    private static final Logger logger = LoggerFactory.getLogger(KubeCluster.class);
+
+    protected void exec(String... cmd) throws KubeClusterException {
+        exec(Arrays.asList(cmd));
+    }
+    protected void exec(List<String> cmd) throws KubeClusterException {
+        try {
+            logger.info("{}", join(" ", cmd));
+            ProcessBuilder pb = new ProcessBuilder(cmd);
+            pb.inheritIO();
+            Process p = pb.start();
+            int sc = p.waitFor();
+            if (sc != 0) {
+                throw new KubeClusterException(sc, cmd + " got status code " + sc);
+            }
+        } catch (IOException e) {
+            throw new KubeClusterException(e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new KubeClusterException(e);
+        }
+    }
+
+    protected String exec2(String... cmd) throws KubeClusterException {
```

Comment:
can we define a common `exec` method with a `boolean redirectOutput` so that if that's true it goes creating the tmp file redirecting output there ? We could avoid to have `exec` and `exec2` in this way. Wdyt ?

Reply:
Yeah, I thought about that, but I don't think it would make the code much better, because in the one case you need to mess around with a temp file (`try/finally`) and in the other you don't. I suppose the perf isn't so important here though, so maybe it doesn't matter if we create a temp file and then don't actually use it for anything. Let me think about this.